### PR TITLE
Add flaubert types

### PR DESCRIPTION
**Pull request by chainyo**
# What does this PR do?

I added type hints for all FlauBERT PyTorch and Tensorflow classes as described in https://github.com/huggingface/transformers/issues/16059 .

@Rocketknight1

### CODE REVIEW

File: src/transformers/models/flaubert/modeling_tf_flaubert.py [hunk 1]
```diff
@@ -827,19 +828,19 @@ def prepare_inputs_for_generation(self, inputs, **kwargs):
...
+        lengths: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        cache: Optional[Dict[str, tf.Tensor]] = None,
+        head_mask: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        inputs_embeds: Optional[tf.Tensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        training: Optional[bool] = False,
         **kwargs,
     ):
```
Comment (Rocketknight1) [inline src/transformers/models/flaubert/modeling_tf_flaubert.py:845]: I think the output type is missing here!

**Review by Rocketknight1 — COMMENTED** [review]
Overall looks great, with one minor nit!

File: src/transformers/models/flaubert/modeling_tf_flaubert.py (hunk 1 above)
Comment (ChainYo) [reply src/transformers/models/flaubert/modeling_tf_flaubert.py]: good point, sorry for forgetting

**Review by Rocketknight1 — APPROVED** [review]
Looks perfect now, thanks for the PR!

**Pull request merged**